### PR TITLE
Feature/comet logging

### PR DESCRIPTION
**Pull request by sherpan**
# Description

Added Comet ML as one of the loggers to choose from.

- Fixes # (issue)

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,19 +242,25 @@ python tools/benchmarking/benchmark.py \
 
 Refer to the [Benchmarking Documentation](https://openvinotoolkit.github.io/anomalib/guides/benchmarking.html) for more details.
 
-# Logging Images
+# Experiment Management 
 
-You can save images locally or to a logger such TensorBoard or Weights and Biases by setting the following configuration.
+Anomablib is integrated with various libraries for experiment tracking such as comet, tensorboard, and wandb through [pytorch lighting loggers](https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html).
```
Comment (Lothiraldan) [inline README.md:247]:
```suggestion
Anomablib is integrated with various libraries for experiment tracking such as Comet, tensorboard, and wandb through [pytorch lighting loggers](https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html). 
```

File: README.md [hunk 2]
```diff
@@ -241,19 +242,25 @@ python tools/benchmarking/benchmark.py \
...
+ logging:
+  logger: [comet, tensorboard, wandb]
+  log_graph: True
 ```
 
-For more information on logging images, refer to the [Logging Documentation](https://openvinotoolkit.github.io/anomalib/guides/logging.html)
+For more information, refer to the [Logging Documentation](https://openvinotoolkit.github.io/anomalib/guides/logging.html)
 
----
+Note: Set your API Key for [comet.ml](https://www.comet.com/signup?utm_source=anomalib&utm_medium=referral) via `comet_ml.init()` in interactive python or simply run `export COMET_API_KEY=<Your API Key>`
```
Comment (Lothiraldan) [inline README.md:263]:
```suggestion
Note: Set your API Key for [Comet.ml](https://www.comet.com/signup?utm_source=anomalib&utm_medium=referral) via `comet_ml.init()` in interactive python or simply run `export COMET_API_KEY=<Your API Key>`
```
Might be worth to link to the following documentation page https://www.comet.com/docs/v2/guides/tracking-ml-training/configuring-comet/

File: anomalib/utils/loggers/comet.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    def add_image(self, image: Union[np.ndarray, Figure], name: Optional[str] = None, **kwargs: Any):
+        """Interface to add image to comet logger.
+
+        Args:
+            image (Union[np.ndarray, Figure]): Image to log
+            name (Optional[str]): The tag of the image
+            kwargs: Accepts only `global_step` (int). The step at which to log the image.
+        """
+        if "global_step" not in kwargs:
+            raise ValueError("`global_step` is required for tensorboard logger")
```
Comment (Lothiraldan) [inline anomalib/utils/loggers/comet.py:108]:
```suggestion
            raise ValueError("`global_step` is required for comet logger")
```

File: requirements/base.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 albumentations>=1.1.0
+comet-ml==3.31.7
```
Comment (Lothiraldan) [inline requirements/base.txt:2]: Why are we pinning the requirements to exactly this version and not greater than? And is there any reason not to support older versions of the Python SDK?

File: docs/source/guides/logging.rst [hunk 5]
```diff
@@ -64,14 +76,19 @@ To log other artifacts to the logger, you can directly access the logger object
   When accessing the base ``logger/logger.experiment`` object, refer to the documentation of the respective logger for the list of available methods.
 
 
-Anomalib makes it easier to log your model graph to TensorBoard or Weights and Biases. Just set ``log_graph`` to True under ``logging`` parameter of the model configuration file.
+Anomalib makes it easier to log your model graph to Comet,TensorBoard or Weights and Biases. Just set ``log_graph`` to True under ``logging`` parameter of the model configuration file.
```
Comment (Lothiraldan) [inline docs/source/guides/logging.rst:79]:
```suggestion
Anomalib makes it easier to log your model graph to Comet, TensorBoard or Weights and Biases. Just set ``log_graph`` to True under ``logging`` parameter of the model configuration file.
```

**Review by Lothiraldan — COMMENTED** [review]
Few small changes and one questions regarding the packaging

File: anomalib/utils/loggers/comet.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            api_key=api_key,
+            save_dir=save_dir,
+            project_name=project_name,
+            rest_api_key=rest_api_key,
+            experiment_name=experiment_name,
+            experiment_key=experiment_key,
+            offline=offline,
+            prefix=prefix,
+            **kwargs
+        )
```
Comment (Lothiraldan) [inline anomalib/utils/loggers/comet.py:95]: Let's add framework tracking so we can follow how many experiment were using the Anomalib integration:
```suggestion
        )
        self.experiment.log_other("Created from", "Anomalib")
```

File: requirements/base.txt (hunk 4 above)
Comment (sherpan) [reply requirements/base.txt]: made it >=!

**Pull request closed without merging**